File: backend/app/Support/ImageUrl.php
<?php

namespace App\Support;

class ImageUrl
{
    /**
     * placehold.co returns SVG by default; mobile clients need a raster format.
     */
    public static function rasterize(?string $url): ?string
    {
        if ($url === null || $url === '') {
            return $url;
        }

        if (! str_contains($url, 'placehold.co/')) {
            return $url;
        }

        if (preg_match('/\.(png|jpe?g|gif|webp|avif)(\?|$)/i', $url)) {
            return $url;
        }

        [$base, $query] = array_pad(explode('?', $url, 2), 2, null);

        return $base.'.png'.($query !== null ? '?'.$query : '');
    }
}

File: chiriqui_flutter/chiriqui_flutter/lib/core/utils/image_url.dart
String? rasterImageUrl(String? url) {
  if (url == null || url.isEmpty) return url;
  if (!url.contains('placehold.co/')) return url;
  if (RegExp(r'\.(png|jpe?g|gif|webp|avif)(\?|$)', caseSensitive: false).hasMatch(url)) {
    return url;
  }

  final parts = url.split('?');
  final base = '${parts.first}.png';
  return parts.length > 1 ? '$base?${parts[1]}' : base;
}
